js/src/jupyter-leaflet.js

```diff
@@ -344,6 +345,45 @@ var LeafletVideoOverlayView = LeafletRasterLayerView.extend({
     },
 });
 
+var LeafletHeatmapView = LeafletLayerView.extend({
+
+    create_obj: function () {
+        this.obj = L.heatLayer(
+            this.model.get('latlngs'),
+            {
+                minOpacity: this.model.get('minOpacity'),
+                maxZoom: this.model.get('maxZoom'),
+                maxIntensity: this.model.get('maxIntensity'),
+                radius: this.model.get('radius'),
+                blur: this.model.get('blur'),
+                gradient: this.model.get('gradient')
+            },
+            this.get_options()
+        );
+    },
+
+    model_events: function () {
+        LeafletHeatmapView.__super__.model_events.apply(this, arguments);
+        this.listenTo(this.model, 'change:latlngs', function () {
+        }, this);
+
+        this.listenTo(this.model, 'change:options', function () {
+            latlngs = this.model.get('latlngs');
+            heat_options = {
+                minOpacity: this.model.get('minOpacity'),
+                maxZoom: this.model.get('maxZoom'),
+                max: this.model.get('maxIntensity'),
+                radius: this.model.get('radius'),
+                blur: this.model.get('blur'),
+                gradient: this.model.get('gradient')
+            }
+            options = this.get_options();
+            this.map_view.obj.removeLayer(this.obj);
+            this.obj = L.heatLayer(latlngs, heat_options, options);
+            this.map_view.obj.addLayer(this.obj);
```

Comment:
Why do you remove the layer and then recreate it? Something like
```javascript
this.obj.setOptions(this.get_options());
```
should do the job